src/macros.jl

```diff
@@ -212,13 +226,34 @@ function emit(loop)
     end
     stmts = Any[]
     append!(stmts, loop.allocations)
+
+    # private_allocations turn into lhs = ntuple(i->rhs, length(__workitems_iterspace()))
+    N = gensym(:N)
+    push!(stmts, :($N = length($__workitems_iterspace())))
+
+    for stmt in loop.private_allocations
+        if @capture(stmt, lhs_ = rhs_)
+            push!(stmts, :($lhs = ntuple(_->$rhs, $N)))
```

Comment:
@simonbyrne you asked if we could support something like this:

```
e = @index(Group, Linear)
i, j = @index(Local, NTuple)
@private mem = Fprog[i,j,:,e]
```

We would need to scan `rhs` for any uses of `loop.indicies` and then project `i`/`_` onto `__workitems_iterspace` inside to obtain a valid `idx` and then instantiate the index expressions `$(loop.indicies...)`. Doable, but pfff.

Reply:
In second thought, I don't think this is really necessary.